download/download_test.go

```diff
@@ -0,0 +1,246 @@
+package download
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+var expectedURLs = []string{
+	"http://200.152.38.155/CNPJ/F.K03200$W.SIMPLES.CSV.D10710.zip",
+	"http://200.152.38.155/CNPJ/F.K03200$Z.D10710.CNAECSV.zip",
+	"http://200.152.38.155/CNPJ/F.K03200$Z.D10710.MOTICSV.zip",
+	"http://200.152.38.155/CNPJ/F.K03200$Z.D10710.MUNICCSV.zip",
+	"http://200.152.38.155/CNPJ/F.K03200$Z.D10710.NATJUCSV.zip",
+	"http://200.152.38.155/CNPJ/F.K03200$Z.D10710.PAISCSV.zip",
+	"http://200.152.38.155/CNPJ/F.K03200$Z.D10710.QUALSCSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y0.D10710.EMPRECSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y0.D10710.ESTABELE.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y0.D10710.SOCIOCSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y1.D10710.EMPRECSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y1.D10710.ESTABELE.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y1.D10710.SOCIOCSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y2.D10710.EMPRECSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y2.D10710.ESTABELE.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y2.D10710.SOCIOCSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y3.D10710.EMPRECSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y3.D10710.ESTABELE.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y3.D10710.SOCIOCSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y4.D10710.EMPRECSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y4.D10710.ESTABELE.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y4.D10710.SOCIOCSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y5.D10710.EMPRECSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y5.D10710.ESTABELE.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y5.D10710.SOCIOCSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y6.D10710.EMPRECSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y6.D10710.ESTABELE.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y6.D10710.SOCIOCSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y7.D10710.EMPRECSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y7.D10710.ESTABELE.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y7.D10710.SOCIOCSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y8.D10710.EMPRECSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y8.D10710.ESTABELE.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y8.D10710.SOCIOCSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y9.D10710.EMPRECSV.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y9.D10710.ESTABELE.zip",
+	"http://200.152.38.155/CNPJ/K3241.K03200Y9.D10710.SOCIOCSV.zip",
+}
+
+func TestGetURLs(t *testing.T) {
+	ts := httpTestServer(t)
+	defer ts.Close()
+
+	got, err := getURLs(ts.Client(), federalRevenue)
+	if err != nil {
+		t.Errorf("Expected getURLs to run withour errors, got: %v:", err)
+		return
+	}
+	assertArraysHaveSameItems(t, got, expectedURLs)
```

Comment:
Did you try to use [reflect.DeepEqual](https://pkg.go.dev/reflect#DeepEqual)?

Reply:
I know about it but I haven't tried. The point is that I just wanna make sure the arrays have the same items (the order is not important). Thus, I assumed the order would matter for `reflect.DeepEqual`…

Reply:
Indeed, the order does matter for `reflect.DeepEqual`.